### ISSUE TITLE
Tweak NN architecture and hyper-parameters

### ISSUE THREAD

**Issue by dzenanz** [OpenImaging/miqa#142]
A refactoring done in #115 for #30 requires new tuning.

**Ideas**
* [x] use deeper network
* [x] use [dropout](https://docs.monai.io/en/latest/_modules/monai/networks/nets/classifier.html) for regularization
* [x] use weight decay via [AdamW](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html)
* [ ] use downscaled image and concatenate with mean over tiles.
* [ ] try residual blocks instead of convolutional blocks

**Comment by dzenanz** [comment]
Increasing the number of convolutional blocks from 4 to 5 slightly improves results.

4 conv blocks:
```text
epoch 5 average loss: 23.0627
confusion matrix:
[[168 164 116  48  16  17   4   4   0   0   0]
 [ 66 313  74   0   0   0   0   0   0   0   0]
 [  1  48 208  48   1   0   0   0   0   0   0]
 [  0   1  53 423  40   0   0   0   0   0   0]
 [  0   0   0  67 367  74   1   0   0   0   0]
 [  0   0   0   1  44 229  49   0   0   0   0]
 [  2   1   3  13  23  65  73  69  46  18  10]
 [  0   0   0   1   3  12  50  80  76  28  13]
 [  0   0   0   1   3   8  37  40  80  57  36]
 [  0   0   0   1   0  10  22  47  96 100  60]
 [  0   1   0   2   7  10  25  47 101  88  74]]

val_confusion_matrix:
[[ 4  3  4 13 12  8  8  2  2  0  1]
 [ 0  0  1  2  0  0  0  0  1  0  0]
 [ 1  0  1  0  1  0  0  1  0  1  0]
 [ 0  0  0  1  0  0  1  0  0  0  0]
 [ 0  1  0  1  1  0  0  0  0  1  0]
 [ 1  0  0  0  1  0  1  0  0  0  0]
 [ 2  5  6 30 43 47 58 49 47 11  7]
 [ 0  0  1  3  3  4  4 12 11  6  3]
 [ 2  0  2  2 14 14 26 45 41 47 21]
 [ 0  0  0  3  4  7 19 36 49 60 45]
 [ 0  1  4  5  8 14 23 30 28 35 21]]

              precision    recall  f1-score   support
           0       0.40      0.07      0.12        57
           1       0.00      0.00      0.00         4
           2       0.05      0.20      0.08         5
           3       0.02      0.50      0.03         2
           4       0.01      0.25      0.02         4
           5       0.00      0.00      0.00         3
           6       0.41      0.19      0.26       305
           7       0.07      0.26      0.11        47
           8       0.23      0.19      0.21       214
           9       0.37      0.27      0.31       223
          10       0.21      0.12      0.16       169
    accuracy                           0.19      1033
   macro avg       0.16      0.19      0.12      1033
weighted avg       0.31      0.19      0.23      1033

current epoch: 5 current metric: -0.08 best metric: 0.02 at epoch 4
```

5 conv blocks:
```text
epoch 4 average loss: 28.0880
confusion matrix:
[[ 87 139 126  70  31  16   8   5   0   0   0]
 [ 78 360  78   1   0   0   0   0   0   0   0]
 [  0  37 217  62   1   0   0   0   0   0   0]
 [  0   2  49 440  44   0   0   0   0   0   0]
 [  0   0   1  66 396  73   0   0   0   0   0]
 [  0   0   0   1  35 178  50   1   0   0   0]
 [  1   2   4  13  27  47  59  71  46  16  11]
 [  0   0   1   0   5  16  34  70  78  50  17]
 [  1   0   3   3  10  10  23  69  72  60  26]
 [  0   0   0   3   4  11  30  57 110  82  46]
 [  0   1   1   4   7  16  36  57  87  68  67]]

val_confusion_matrix:
[[ 2  3  6  9 14 10  8  1  1  2  1]
 [ 0  0  2  1  0  0  0  0  0  1  0]
 [ 1  0  1  0  0  1  0  0  1  0  1]
 [ 0  0  0  0  1  0  1  0  0  0  0]
 [ 0  0  1  1  0  1  0  0  1  0  0]
 [ 1  0  0  1  0  0  0  1  0  0  0]
 [ 2  3  9 16 41 53 68 59 34 15  5]
 [ 0  0  2  2  1  4  5 11 10  9  3]
 [ 1  2  0  4  5 12 26 30 63 47 24]
 [ 0  0  1  1  4  6 15 28 50 67 51]
 [ 0  0  0  9  5 12 22 37 25 34 25]]

              precision    recall  f1-score   support
           0       0.29      0.04      0.06        57
           1       0.00      0.00      0.00         4
           2       0.05      0.20      0.07         5
           3       0.00      0.00      0.00         2
           4       0.00      0.00      0.00         4
           5       0.00      0.00      0.00         3
           6       0.47      0.22      0.30       305
           7       0.07      0.23      0.10        47
           8       0.34      0.29      0.32       214
           9       0.38      0.30      0.34       223
          10       0.23      0.15      0.18       169
    accuracy                           0.23      1033
   macro avg       0.17      0.13      0.12      1033
weighted avg       0.35      0.23      0.27      1033

current epoch: 4 current metric: 0.02 best metric: 0.02 at epoch 4
```

**Comment by dzenanz** [comment]
Adding dropout noticeably improves results, especially on the small set. Here is 5 conv blocks, and `dropout = 0.1` on the large set:
```text
epoch 14 average loss: 34.8976
confusion matrix:
[[ 68  92 145  95  75  24  11   3   0   0   0]
 [ 90 251 121  14   0   0   0   0   0   0   0]
 [  5  56 195  70   9   0   0   0   0   0   0]
 [  0   8  99 285 119   2   0   0   0   0   0]
 [  0   0  12 113 224 137  14   0   0   0   0]
 [  0   0   1   4  58 193  64   4   1   0   0]
 [  0   2   4  17  30  53  75  49  36  22   9]
 [  0   0   0   3  17  23  42  55  71  53  13]
 [  0   1   3   2   7  15  21  51  64  56  23]
 [  0   0   2   4   3  13  32  65 104  74  56]
 [  0   0   2   6  12  34  45  73  73  63  44]]

val_confusion_matrix:
[[ 5 10  9 11  3  6  7  2  0  3  1]
 [ 0  1  0  0  1  0  1  0  1  0  0]
 [ 0  1  1  0  1  0  0  1  1  0  0]
 [ 0  0  1  0  1  0  0  0  0  0  0]
 [ 0  0  1  1  1  0  0  0  0  0  1]
 [ 0  0  0  0  1  0  1  0  1  0  0]
 [ 1  1  3 13 26 41 53 69 53 35 10]
 [ 0  0  0  0  1  5  4 15  9 11  2]
 [ 0  0  1  3  6 16 19 28 66 46 29]
 [ 0  0  0  0  2 12  9 36 48 68 48]
 [ 0  0  1  4  4 14 20 31 40 36 19]]
              precision    recall  f1-score   support
           0       0.83      0.09      0.16        57
           1       0.08      0.25      0.12         4
           2       0.06      0.20      0.09         5
           3       0.00      0.00      0.00         2
           4       0.02      0.25      0.04         4
           5       0.00      0.00      0.00         3
           6       0.46      0.17      0.25       305
           7       0.08      0.32      0.13        47
           8       0.30      0.31      0.30       214
           9       0.34      0.30      0.32       223
          10       0.17      0.11      0.14       169
    accuracy                           0.22      1033
   macro avg       0.21      0.18      0.14      1033
weighted avg       0.35      0.22      0.25      1033
saved new best metric model as /home/dzenan/miqa/miqa/learning/models/miqaT1-val0.pth
current epoch: 14 current metric: 0.12 best metric: 0.12 at epoch 14
Learning rate after epoch 14: 2.0589113209464913e-05
```

**Comment by dzenanz** [comment]
A more aggressive dropout of 0.5 works less well.

**Comment by dzenanz** [comment]
Using AdamW also slightly improves results.
